### DISCUSSION TITLE
How to change CI/CD provider during project creation?

### DISCUSSION THREAD

**Discussion by lucascanna** [mia-platform/community#332] [Q&A] [answered]
### Which section?

Console

### Which is the version?

11.7.1

### Description

Hi! 

During project creation, in the "Environments" step, both the CI/CD provider and the Secret Manager provider are already set and it is not possible to change them. How can I set a different CI/CD provider for my project?

**Comment by davidebianchi** [comment]
Hi! At the moment, it is possible to change it only through the CMS after the project creation. 
It is possible to change the default used provider for the company in the project creation ([here](https://docs.mia-platform.eu/docs/console/company-configuration/providers/management#set-a-default-provider) the docs).

We are now adding some other provider configuration in the Project creation, if I correctly remember the configuration of the CI/CD tool. @fredmaggiowski have you more info about it?

**Reply by fredmaggiowski** [reply, marked as answer]
Yes! @Lucascanna I just merged a PR adding the selection for the CI/CD provider during project creation; if this sounds helpful we may also add the Secret Manager provider in the near future.

**Reply by phil-hotchkiss** [reply]
@fredmaggiowski Is it possible to update a provider after the project has been created? For example, maybe I'd want to consider changing the `Secret Manager` from Gitlab to Vault without creating a new project.

**Reply by fredmaggiowski** [reply]
Hi @phil-hotchkiss right now changing providers for an existing project is only possible using the console backoffice application.

In order to change the secret manager your can follow this [documentation page](https://docs.mia-platform.eu/docs/console/company-configuration/providers/project-configuration#edit-a-git-provider). If you feel like you might need further assistance don't hesitate to contact your Mia-Platform referent.
